app/src/androidTest/java/com/android/streetworkapp/ui/map/MapTest.kt

```diff
@@ -0,0 +1,59 @@
+package com.android.streetworkapp.ui.map
+
+import android.Manifest
+import androidx.activity.result.ActivityResultLauncher
+import androidx.compose.ui.test.assertIsDisplayed
+import androidx.compose.ui.test.junit4.createComposeRule
+import androidx.compose.ui.test.onNodeWithTag
+import androidx.compose.ui.test.onNodeWithText
+import androidx.compose.ui.test.performClick
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.mockito.Mockito.mock
+import org.mockito.kotlin.times
+import org.mockito.kotlin.verify
+
+class MapTest {
+
+  @get:Rule val composeTestRule = createComposeRule()
+
+  private lateinit var requestPermissionLauncher: ActivityResultLauncher<String>
+
+  @Before
+  fun setup() {
+    // Mock the ActivityResultLauncher using Mockito
+    requestPermissionLauncher = mock()
+  }
+
+  @Test
+  fun buttonClick() {
+    // Set the content of the activity
+    composeTestRule.setContent { MapManager(requestPermissionLauncher = requestPermissionLauncher) }
+
+    // The first request
+    composeTestRule.onNodeWithTag("requestPermissionButton").assertIsDisplayed()
+    verify(requestPermissionLauncher, times(1)).launch(Manifest.permission.ACCESS_FINE_LOCATION)
+    // The first request second request if pressed
+    composeTestRule.onNodeWithTag("requestPermissionButton").performClick()
+    composeTestRule.onNodeWithText("Test Permission").assertIsDisplayed()
+
+    // Verify that the launch method of requestPermissionLauncher was called with the correct
+    verify(requestPermissionLauncher, times(2)).launch(Manifest.permission.ACCESS_FINE_LOCATION)
+  }
+
+  @Test
+  fun buttonNotClick() {
+    verify(requestPermissionLauncher, times(0)).launch(Manifest.permission.ACCESS_FINE_LOCATION)
+    // Set the content of the activity
+    composeTestRule.setContent { MapManager(requestPermissionLauncher = requestPermissionLauncher) }
+
+    // The first request
+    composeTestRule.onNodeWithTag("requestPermissionButton").assertIsDisplayed()
+    verify(requestPermissionLauncher, times(1)).launch(Manifest.permission.ACCESS_FINE_LOCATION)
+    composeTestRule.onNodeWithText("Test Permission").assertIsDisplayed()
+
+    // Verify that the launch method of requestPermissionLauncher was called with the correct
+    verify(requestPermissionLauncher, times(1)).launch(Manifest.permission.ACCESS_FINE_LOCATION)
```

Comment:
There is many uses of `Manifest.permission.ACCESS_FINE_LOCATION` at line 36, 42, 47, 53 and 57. 
Is it possible to make them into a constant to reduce error.